examples/systemcalls/systemcalls.c

```diff
@@ -58,10 +72,32 @@ bool do_exec(int count, ...)
  *   as second argument to the execv() command.
  *
 */
+    pid_t pid;
+    pid = fork();
 
-    va_end(args);
-
-    return true;
+    if (pid == -1) {
+        perror("Error executing fork");
+        va_end(args);
+        return false;
+    } else if (pid == 0) {
+        // Child process.Calling execv
+        execv(command[0], command);
+        // If execv fails
+	perror("Error executing execv"); 
+	exit(EXIT_FAILURE);
+    } else {
+        // Parent process
+        int status;
+        waitpid(pid, &status, 0);
+        // check status of waitpid
+        if (WIFEXITED(status) && WEXITSTATUS(status) == 0) {
+            va_end(args);
+            return true;
+        } else {
+            va_end(args);
+            return false;
+        }
+    }
```

Comment:
good idea to cleanup before inline returns, but I'd suggest structuring the function such that there's an exit at the end.